tuf/api/metadata.py

```diff
@@ -266,6 +269,63 @@ def sign(
 
         return signature
 
+    def verify_delegate(
+        self,
+        delegated_role: str,
+        delegated_metadata: "Metadata",
+        signed_serializer: Optional[SignedSerializer] = None,
+    ) -> None:
+        """Verifies that 'delegate' is signed with the required threshold of
+        keys for the delegated role 'role_name'.
+
+        Args:
+            delegated_role: Name of the delegated role to verify
+            delegated_metadata: The Metadata object for the delegated role
+            signed_serializer: Optional; serializer used for delegate
+                serialization. Default is CanonicalJSONSerializer.
+
+        Raises:
+            UnsignedMetadataError: 'delegate' was not signed with required
+                threshold of keys for 'role_name'
+        """
+
+        # Find the keys and role in delegator metadata
+        role = None
+        if isinstance(self.signed, Root):
+            keys = self.signed.keys
+            role = self.signed.roles.get(delegated_role)
+        elif isinstance(self.signed, Targets):
+            if self.signed.delegations:
```

Comment:
Nit: I wonder if this would be slightly clearer if we did something like:
```Python
if not self.signed.delegations:
    raise TypeError("Targets does not delegate to any other roles")
keys = ...
...
```
?